feature/AP5-19

dmstringer opened the pull request.

Commit:
feature/AP5-19

aaronpaulroberts commented on rigex-front-end/src/pages/Landing/Landing.js:
```diff
@@ -6,6 +6,10 @@ import NavBar from '../../components/navBar/NavBar';
+import RigSidebar from '../../components/RigSidebar/RigSidebar';
+import '../../styles/styles.scss';
+
+let listOfRigs = [];
```
why is this list scoped to the file?
Is there a better place to scope your rig list? Are you using redux? Or even just scoped as an attribute on the component function?

dmstringer replied:
Now that the PR for the setup of the rigs query and rigs redux store has been submitted I was able to create and test a proper listOfRigs. This and the other comments are addressed is my next commit.

aaronpaulroberts commented on rigex-front-end/src/components/RigSidebar/RigList.js:
```diff
@@ -0,0 +1,33 @@
+
+import './sideBar.scss';
+
+function RigList(props) {
```
is there a particular reason why this isn't an arrow function?
If there isn't, please update this to be an arrow function

aaronpaulroberts replied:
also, you can destructure `props` here to be `({ rigList })`. That way you can just call `rigList` instead of `props.rigList`

you seem to be using rigList twice, but nothing else from props nor are you passing the whole props object to a child, so you don't need the whole object after this point

aaronpaulroberts commented on rigex-front-end/src/components/RigSidebar/RigList.js:
```diff
@@ -0,0 +1,33 @@
+    setSelectedIndex(index);
+  };
+
+  props.rigList.sort((a, b) => (a.name > b.name ? 1 : -1));
```
This will only sort on initial component render. If there is behavior that would necessitate a resort throughout the lifetime of the component, it won't happen because of the current scope

aaronpaulroberts replied:
https://reactjs.org/docs/hooks-reference.html#useeffect

aaronpaulroberts commented on rigex-front-end/src/assets/Tower@1.5x.svg:
```diff
@@ -0,0 +1,11 @@
+            <line x1="1.22247048e-12" y1="76.6915657" x2="43.0118561" y2="76.6915657" id="Path-2" stroke="#98A8B1" stroke-width="2.25" stroke-linecap="round"></line>
+        </g>
+    </g>
+</svg>
```
please add a newline at the end of this file (not that important, but it removes this little lint warning/error

aaronpaulroberts commented on rigex-front-end/src/components/RigSidebar/EmptyCard.js:
```diff
@@ -0,0 +1,18 @@
+import emptyRigs from '../../assets/Tower@1.5x.svg';
+import './sideBar.scss';
+
+function EmptyCard() {
```
please change this to an arrow function

aaronpaulroberts commented on rigex-front-end/src/components/RigSidebar/RigSidebar.js:
```diff
@@ -0,0 +1,29 @@
+import EmptyCard from '../../components/RigSidebar/EmptyCard';
+import './sideBar.scss';
+
+function RigSidebar(props) {
```
please see previous destructure and arrow function comments

Commit:
feature/AP5-19: correcting issues from feedback

dmstringer committed:
Update Landing.js

After discussion with other apprentices, we felt it better assign listOfRigs from the store instead of directly from the query

BarretteBanner commented on rigex-front-end/src/components/RigSidebar/EmptyCard.js:
```diff
@@ -0,0 +1,18 @@
+
+const EmptyCard = () => {
+  return (
+    <Box className='sidebarBox'>
```
can we use regular scss for this.

BarretteBanner commented on rigex-front-end/src/components/RigSidebar/RigList.js:
```diff
@@ -0,0 +1,35 @@
+  });
+
+  return (
+    <List>
```
I think all of this can also be regular scss and html

Commit:
feature/AP5-19: requested comversion of material-ui items
